### PR TITLE
Developer section reorg + Updated custom check docs

### PR DESCRIPTION
**Pull request by apigirl**
### What does this PR do?

Reorgs the Developer section and adds context about several ways to programmatically send data to Datadog.

### Motivation

Datadog as a platform!

### Preview

https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/

---

### Reviewer checklist

- [ ] Click on every nav item under https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/ to make sure they are all live.
- [ ] Double check that the single source file that was moved to https://docs-staging.datadoghq.com/kaylyn/dev-toc/integrations/guide/amazon_cloudformation/ is live
- [ ] Double check that the new Events guide landing page and guides are live - https://docs-staging.datadoghq.com/kaylyn/dev-toc/events/guides/
- [ ] Review the new and heavily updated pages:
     - [ ] https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/
     - [ ] https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/custom_checks/
     - [ ] https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/custom_checks/write_agent_check/
     - [ ] https://docs-staging.datadoghq.com/kaylyn/developers/service_checks/

### To-do
- [x] aliases
- [x] figure out why the [cloudformation file](https://github.com/DataDog/documentation/pull/10864/files#diff-db2c7d7ed4ad74540d6be20f41ed13a6c933a36810d127e87f033fdf76ec3e0a) is showing up in the PR 
- [x] figure out why the email guide isn't working - https://docs-staging.datadoghq.com/kaylyn/dev-toc/events/guides/
- [x] figure out why the nav item Service Check > Submission API isn't working
- [x] double check the custom check and service check info in the main landing page

### CODE REVIEW

File: content/en/developers/service_checks/_index.md [hunk 1]
```diff
@@ -10,21 +10,24 @@ aliases:
...
-| Status Code | Description |
-| ----------- | ----------- |
-| `0`         | OK          |
-| `1`         | Warning     |
-| `2`         | Critical    |
-| `3`         | Unknown     |
+Service checks at the cluster level offer another effective way to monitor distributed or redundant systems that can withstand some failures. Use these alerts for architectures where individual hosts run multiple services, because they can surface the degradation of the service even if the hosts running that service remain available (and would pass a host-level health check).
 
-{{< whatsnext desc="Find below how to submit a Service Check to Datadog:">}}
+You can set up monitoring and an alert for when a critical, non-redundant service is lost, or if a cluster is on the verge of failure due to widespread node loss. Other critiacl alerts could be a drop in request throughput or an increase in request latency.
```
Comment (sarina-dd) [inline content/en/developers/service_checks/_index.md:17]:
```suggestion
You can set up monitoring and an alert for when a critical, non-redundant service is lost, or if a cluster is on the verge of failure due to widespread node loss. Other critical alerts could be a drop in request throughput or an increase in request latency.
```

File: content/en/developers/service_checks/_index.md [hunk 2]
```diff
@@ -10,21 +10,24 @@ aliases:
...
+
+You might need to set up a service check if an integration does not have one natively or for an internal service that you want to see up or down status.
```
Comment (sarina-dd) [inline content/en/developers/service_checks/_index.md:19]:
```suggestion
You might need to set up a service check if an integration does not have one natively or for an internal service that you want to monitor for up or down status.
```

File: content/en/events/guides/dogstatsd.md [hunk 3]
```diff
@@ -6,12 +6,14 @@ further_reading:
...
   text: "Introduction to DogStatsD"
-- link: "/developers/libraries/"
+- link: "/developers/community/libraries/"
   tag: "Documentation"
   text: "Official and Community created API and DogStatsD client libraries"
 - link: "https://github.com/DataDog/datadog-agent/tree/master/pkg/dogstatsd"
   tag: "GitHub"
   text: "DogStatsD source code"
+aliases:
+    - /events/guides/dogstatsd/
```
Comment (sarina-dd) [inline content/en/events/guides/dogstatsd.md:16]:
```suggestion
    - /developers/events/dogstatsd/
```

File: content/en/events/guides/email.md [hunk 4]
```diff
@@ -2,7 +2,8 @@
 title: Events with email
 kind: documentation
 aliases:
-- /guides/eventsemail
+    - /events/guides/email/
```
Comment (sarina-dd) [inline content/en/events/guides/email.md:5]:
```suggestion
    - /developers/events/email/
```

File: content/en/developers/community/office_hours.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 ---
 title: Community Office Hours
 type: documentation
+aliases:
+   - /developers/community/office_hours/
```
Comment (sarina-dd) [inline content/en/developers/community/office_hours.md:5]:
```suggestion
   - /developers/office_hours/
```

File: content/en/events/guides/_index.md [hunk 6]
```diff
@@ -8,14 +8,16 @@ further_reading:
 - link: "/monitors/monitor_types/event/"
   tag: "Documentation"
   text: "Event monitors"
+aliases:
+    - /events/guides/
```
Comment (sarina-dd) [inline content/en/events/guides/_index.md:12]:
```suggestion
    - /events/
```

File: content/en/events/guides/agent.md [hunk 7]
```diff
@@ -2,9 +2,11 @@
 title: Events with a Custom Agent Check
 kind: documentation
 further_reading:
-- link: "/developers/write_agent_check/"
+- link: "/developers/custom_checks/write_agent_check/"
   tag: "Documentation"
   text: "Writing a Custom Agent Check"
+aliases:
+    - /events/guides/agent/
```
Comment (sarina-dd) [inline content/en/events/guides/agent.md:9]:
```suggestion
    - /events/agent/
```

File: content/en/developers/community/libraries.md [hunk 8]
```diff
@@ -4,6 +4,7 @@ kind: documentation
 aliases:
    - /libraries/
    - /developers/faq/monitoring-akka/
+   - /developers/community/libraries/
```
Comment (sarina-dd) [inline content/en/developers/community/libraries.md:7]:
```suggestion
   - /developers/libraries/
```

**Review by sarina-dd — COMMENTED** [review]
Review is complete @apigirl! 👍

File: content/en/api/latest/_index.md [hunk 9]
```diff
@@ -201,5 +201,5 @@ Trying to get started with the application instead? Check out our general [Getti
 [3]: https://run.pstmn.io/button.svg
 [4]: /api/v1/using-the-api/
 [5]: https://brew.sh
-[6]: https://docs.datadoghq.com/developers/libraries/
+[6]: https://docs.datadoghq.com/developers/community/libraries/
```
[the remark this answers is not in the document]
Comment (apigirl) [reply content/en/api/latest/_index.md:204]: right, this isn't live yet, it's a change made in this PR - https://docs-staging.datadoghq.com/kaylyn/dev-toc/developers/community/libraries/

File: content/en/developers/_index.md [hunk 10]
```diff
@@ -93,30 +99,12 @@ If you are a partner developing for the Datadog Marketplace or community integra
...
-[29]: https://www.datadoghq.com/pricing/
+[3]: /developers/dogstatsd/
+[4]: /developers/custom_checks/write_agent_check/
+[5]: /developers/custom_checks/prometheus/
+[6]: /developers/integrations/
+[7]: https://github.com/DataDog/integrations-extras
+[8]: https://www.datadoghq.com/pricing/
+[9]: /developers/marketplace/
+[10]: /developers/custom_checks/
+[11]: /developers/service_checks/
```
Comment (apigirl) [inline content/en/developers/_index.md:110]:
```suggestion
[11]: /developers/service_checks/
[12]: https://github.com/DataDog/integrations-core
```

File: content/en/developers/community/_index.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
+---
+title: Community
+kind: documentation
+---
+
+Datadog loves our awesome open source community! We wouldn't be able to do it without you.
```
[the remark this answers is not in the document]
Comment (apigirl) [reply content/en/developers/community/_index.md:6]: haha yeah, I would like to leave this one as-is

File: content/en/developers/custom_checks/prometheus.md [hunk 12]
```diff
@@ -13,6 +13,7 @@ further_reading:
   text: "Create a new Integration"
 aliases:
   - /developers/openmetrics/
+  - /developers/custom_checks/prometheus/
```
[the remark this answers is not in the document]
Comment (apigirl) [reply content/en/developers/custom_checks/prometheus.md:16]: great catches! I messed ALL of these up 🤦‍♀️

File: content/en/tracing/profiler/enabling.md [hunk 13]
```diff
@@ -88,11 +88,11 @@ The following profiling features are available depending on your Python version:
 
 |      Feature         | Supported Python versions          |
 |----------------------|------------------------------------|
-| Wall time profiling  | Python >= 2.7                      |
-| CPU time profiling   | Python >= 2.7 on POSIX platforms   |
-| Exception profiling  | Python >= 3.7 on POSIX platforms   |
-| Lock profiling       | Python >= 2.7                      |
-| Memory profiling     | Python >= 3.5                      |
+| Wall time profiling  | Python >= 2.7                      |
+| CPU time profiling   | Python >= 2.7 on POSIX platforms   |
```
Comment (apigirl) [inline content/en/tracing/profiler/enabling.md:92]: not sure what changed here either

File: content/ja/getting_started/site/_index.md [hunk 14]
```diff
@@ -5,8 +5,8 @@ kind: documentation
 
 Datadog では、世界中でさまざまなサイトを提供しています。各サイトは完全に独立しており、サイト間でデータを共有することはできません。各サイトを使用する利点があり（政府のセキュリティ規定など）、世界中の特定した場所にデータを保存することが可能になります。
 
-| サイト    | サイト URL                   | 場所 |
-|---------|----------------------------|----------|
+| サイト    | サイト URL                 | 場所 |
```
Comment (apigirl) [inline content/ja/getting_started/site/_index.md:8]: this is still working, so not going to change

File: content/ja/logs/indexes/_index.md [hunk 15]
```diff
@@ -161,4 +161,4 @@ Log Explorer は、[複数のインデックスにわたるクエリ][7]をサ
 [17]: /ja/help/
 [18]: /ja/logs/live_tail/#overview
 [19]: https://www.timeanddate.com/worldclock/converter.html
-[20]: /ja/logs/guide/logs-monitors-on-volumes/#monitor-indexed-logs-with-fixed-threshold
\ No newline at end of file
+[20]: /ja/logs/guide/logs-monitors-on-volumes/#monitor-indexed-logs-with-fixed-threshold
```
Comment (apigirl) [inline content/ja/logs/indexes/_index.md:164]: this looks the same too

File: content/ja/metrics/advanced-filtering.md [hunk 16]
```diff
@@ -85,4 +85,4 @@ sum:kubernetes.pods.running{service:*-canary} by {service}
 
 ## その他の参考資料
 
-{{< partial name="whats-next/whats-next.html" >}}
+{{< partial name="whats-next/whats-next.html" >}}
```
Comment (apigirl) [inline content/ja/metrics/advanced-filtering.md:88]: also the same

File: data/api/v1/full_spec.yaml [hunk 17]
```diff
@@ -25197,4 +25197,4 @@ tags:
   externalDocs:
     url: https://docs.datadoghq.com/integrations/webhooks
   name: Webhooks Integration
-x-group-parameters: true
+x-group-parameters: true
```
Comment (apigirl) [inline data/api/v1/full_spec.yaml:25200]: also the same

File: .vale.ini [hunk 18]
```diff
@@ -16,4 +16,4 @@ Datadog.latin = NO
 Datadog.gender = NO
 Datadog.emdash = NO
 Datadog.dashes = NO
-Datadog.americanspelling = NO
+Datadog.americanspelling = NO
```
Comment (apigirl) [inline .vale.ini:19]: I don't think this is a real change

File: content/en/monitors/notifications.md [hunk 19]
```diff
@@ -561,5 +561,5 @@ If `host.name` matches `<HOST_NAME>`, the template outputs:
 [11]: /integrations/#cat-notification
 [12]: /events/
 [13]: /monitors/guide/template-variable-evaluation/
-[14]: /monitors/faq/what-are-recovery-thresholds/
-[15]: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+[14]: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
```
Comment (apigirl) [inline content/en/monitors/notifications.md:564]: I think this was the link precommit hook, but it is correct

File: content/ja/real_user_monitoring/_index.md [hunk 20]
```diff
@@ -69,11 +69,12 @@ Datadog のリアルユーザーモニタリング (RUM) は、個々のユー
...
+
 ## その他の参考資料
 
 {{< partial name="whats-next/whats-next.html" >}}
 
 [1]:/ja/real_user_monitoring/dashboards
 [2]:/ja/real_user_monitoring/explorer/analytics
-[4]:/ja/real_user_monitoring/error_tracking
-[1]: /ja/real_user_monitoring/connect_rum_and_traces
+[3]:/ja/real_user_monitoring/error_tracking
```
Comment (apigirl) [inline content/ja/real_user_monitoring/_index.md:79]: this is the precommit hook, but it is correct

File: config/_default/menus/menus.en.yaml [hunk 21]
```diff
@@ -1046,25 +1051,65 @@ main:
     identifier: metrics_top_level
```
Comment (jtappa) [inline config/_default/menus/menus.en.yaml:1051]: This is just missing Metrics Units

File: .vale.ini (hunk 18 above)
Comment (sarina-dd) [reply .vale.ini:19]: Just removed a newline at the end of file, all good 👍

File: content/en/tracing/profiler/enabling.md (hunk 13 above)
Comment (sarina-dd) [reply content/en/tracing/profiler/enabling.md:92]: Interesting, looks fine to me 👍

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**